owsi-core/owsi-core-components/owsi-core-component-jpa-more/src/main/java/fr/openwide/core/jpa/more/business/history/service/IAbstractHistoryLogService.java

```diff
@@ -0,0 +1,48 @@
+package fr.openwide.core.jpa.more.business.history.service;
+
+import java.util.Date;
+import java.util.List;
+
+import fr.openwide.core.jpa.business.generic.service.IGenericEntityService;
+import fr.openwide.core.jpa.exception.SecurityServiceException;
+import fr.openwide.core.jpa.exception.ServiceException;
+import fr.openwide.core.jpa.more.business.difference.service.IDifferenceService;
+import fr.openwide.core.jpa.more.business.difference.util.IDifferenceFromReferenceGenerator;
+import fr.openwide.core.jpa.more.business.difference.util.IHistoryDifferenceGenerator;
+import fr.openwide.core.jpa.more.business.history.model.AbstractHistoryLog;
+import fr.openwide.core.jpa.more.business.history.model.AbstractHistoryDifference;
+import fr.openwide.core.jpa.more.business.history.model.bean.AbstractHistoryLogObjectsBean;
+import fr.openwide.core.jpa.more.business.history.util.IDifferenceHandler;
+
+public interface IAbstractHistoryLogService<
```

Comment:
This is the pattern we use for this sort of thing (you can already find several occurrences of it in the core). Even if I agree it's not particularly accurate.